ql/src/java/org/apache/hadoop/hive/ql/lockmgr/HiveLockObject.java

```diff
@@ -85,7 +85,9 @@ public HiveLockObjectData(String data) {
       lockMode = elem[2];
       queryStr = StringInternUtils.internIfNotNull(elem[3]);
       if (elem.length >= 5) {
-        clientIp = elem[4];
+        String prefix = queryId + ":" + lockTime + ":" + lockMode + ":" + queryStr + ":";
+        // The client IP is the suffix after the 4th colon, this can be IPv6 with colons
+        clientIp = data.substring(data.indexOf(prefix) + prefix.length());
```

Comment:
```
clientIp = data.substring(prefix.length());
```
`data.indexOf(prefix) ` this would be 0 always, If I am catching it right

Reply:
Right, it is always 0.